opentelemetry-exporter-geneva/geneva-uploader/src/config_service/mod.rs

```diff
@@ -0,0 +1,329 @@
+                    ..
+                } => {
+                    assert_eq!(status, 403);
+                }
+                _ => panic!("Expected RequestFailed error, got: {:?}", err),
+            },
+            _ => panic!("Expected error, got success"),
+        }
+    }
+
+    #[cfg_attr(target_os = "macos", ignore)] // cert generated not compatible with macOS
+    #[tokio::test]
+    async fn test_missing_ingestion_gateway_info() {
+        let mock_server = MockServer::start().await;
+
+        // Response without IngestionGatewayInfo
+        let mock_response = serde_json::json!({
+            "SomeOtherField": "value"
+        });
+
+        Mock::given(method("GET"))
+            .and(path(
+                "/api/agent/v3/mockenv/mockacct/MonitoringStorageKeys/",
+            ))
+            .respond_with(ResponseTemplate::new(200).set_body_json(mock_response))
+            .mount(&mock_server)
+            .await;
+
+        let (temp_p12_file, password) = generate_self_signed_p12();
+
+        let config = GenevaConfigClientConfig {
+            endpoint: mock_server.uri(),
+            environment: "mockenv".into(),
+            account: "mockacct".into(),
+            namespace: "mockns".into(),
+            region: "mockregion".into(),
+            config_major_version: 1,
+            auth_method: AuthMethod::Certificate {
+                path: PathBuf::from(temp_p12_file.path().to_string_lossy().to_string()),
+                password,
+            },
+        };
+
+        let client = GenevaConfigClient::new(config).unwrap();
+        let result = client.get_ingestion_info().await;
+
+        assert!(result.is_err());
+        match result {
+            Err(err) => match err {
+                crate::config_service::client::GenevaConfigClientError::AuthInfoNotFound(_) => {
+                    // Test passed
+                }
+                _ => panic!("Expected AuthInfoNotFound error, got: {:?}", err),
+            },
+            _ => panic!("Expected error, got success"),
+        }
+    }
+
+    #[cfg_attr(target_os = "macos", ignore)] // cert generated not compatible with macOS
+    #[tokio::test]
+    async fn test_invalid_certificate_path() {
+        let config = GenevaConfigClientConfig {
+            endpoint: "https://example.com".to_string(),
+            environment: "env".to_string(),
+            account: "acct".to_string(),
+            namespace: "ns".to_string(),
+            region: "region".to_string(),
+            config_major_version: 1,
+            auth_method: AuthMethod::Certificate {
+                path: PathBuf::from("/nonexistent/path.p12".to_string()),
+                password: "test".to_string(),
+            },
+        };
+
+        let result = GenevaConfigClient::new(config);
+
+        assert!(result.is_err());
+        match result {
+            Err(err) => match err {
+                crate::config_service::client::GenevaConfigClientError::Io(_) => {
```

Comment:
I think returning a `CertificateError` instead of `Io` would be more intuitive. In fact, we could only keep `CertificateError` variant and remove the `Io` variant. I believe reading the certificate was the only place where we could have returned the `Io` error.